sycl-fusion/common/include/DynArray.h

```diff
@@ -84,6 +85,28 @@ template <typename T> class DynArray {
   }
 };
 
+///
+/// String-like class that owns its character storage and provides implicit
+/// conversions from and to C strings.
+class DynString {
+public:
+  DynString(const char *Str) {
+    size_t Len = std::strlen(Str);
+    Chars = DynArray<char>(Len + 1);
+    std::copy(Str, Str + Len, Chars.begin());
+    Chars[Len] = '\0';
+  }
+
+  operator const char *() const { return Chars.begin(); }
```

Comment:
```suggestion
  const char *c_str() const { return Chars.begin(); }
```
Implicit conversions can be troublesome. Also, as this does not transfer ownership, just returns a pointer, I don't think a conversion is ideal